bot-twitter/src/keeper.ts

```diff
@@ -0,0 +1,101 @@
+import { AuctionInitialInfo } from 'auctions-core/src/types';
+import getSigner, { createSigner, setSigner } from 'auctions-core/src/signer';
+import { bidOnTheAuction, enrichAuction } from 'auctions-core/src/auctions';
+import {
+    authorizeCollateral,
+    authorizeWallet,
+    getCollateralAuthorizationStatus,
+    getWalletAuthorizationStatus,
+} from 'auctions-core/src/authorizations';
+import { ETHEREUM_NETWORK, KEEPER_MIN_PROFIT_DAI, KEEPER_WALLET_PRIVATE_KEY } from './variables';
+import { setupWallet } from '~/src/authorizations';
+
+function areKeeperVariablesSet(): boolean {
+    if (!KEEPER_WALLET_PRIVATE_KEY) {
+        return false;
+    }
+    return Number.isNaN(KEEPER_MIN_PROFIT_DAI);
+}
+
+export async function setupKeeper() {
+    if (!areKeeperVariablesSet()) {
+        console.warn('keeper: keeper variables are not set. Please check configuration.');
+        return;
+    }
+    if (KEEPER_WALLET_PRIVATE_KEY) {
+        setSigner(ETHEREUM_NETWORK, createSigner(ETHEREUM_NETWORK, KEEPER_WALLET_PRIVATE_KEY));
+        await setupWallet(ETHEREUM_NETWORK);
+    }
```

Comment:
Here, after the wallet is configured, you can set:
```suggestion
        await setupWallet(ETHEREUM_NETWORK);
    }
    isSetupCompleted = true;
```

Otherwise, as explained above, if `setupWallet` fails, `participate` function would still engage and throw errors for every auction. You can test this edge case by providing invalid `KEEPER_WALLET_PRIVATE_KEY`

Reply:
Where do I check for `isSetupCompleted`? I assume in the participate function.

Reply:
Yes!